src/rapids_dependency_file_generator/schema.json

```diff
@@ -0,0 +1,160 @@
+ {
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/rapidsai/dependency-file-generator/v1.0.0/src/rapids_dependency_file_generator/schema.json",
+    "type": "object",
+    "title": "RAPIDS Package Dependency Specification Format",
+    "description": "Consolidated specification of RAPIDS project dependencies.",
+    "properties": {
+        "files": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "type": "object",
+                    "properties": {
+                        "output": {"$ref": "#/$defs/Outputs"},
+                        "includes": {"type": "array", "items": {"type": "string"}},
+                        "matrix": {"$ref": "#/$defs/Matrix"},
+                        "requirements_dir": {"type": "string"},
+                        "conda_dir": {"type": "string"}
+                    },
+                    "additionalProperties": false,
+                    "required": ["output", "includes"]
+                }
+            }
+        },
+        "dependencies": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "type": "object",
+                    "properties": {
+                        "common": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "output_types": {"$ref": "#/$defs/Outputs"},
+                                    "packages": {"$ref": "#/$defs/Packages"}
+                                },
+                                "required": ["output_types", "packages"],
+                                "additionalProperties": false
+                            }
+                        },
+                        "specific": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "output_types": {"$ref": "#/$defs/Outputs"},
+                                    "matrices": {"$ref": "#/$defs/Matrices"}
+                                },
+                                "required": ["output_types", "matrices"],
+                                "additionalProperties": false
+                            }
+                        }
+                    },
+                    "minProperties": 1,
+                    "additionalProperties": false
+                }
+            }
+        },
+        "channels": {"$ref": "#/$defs/Channels"}
+    },
+    "required": ["files", "dependencies"],
+    "additionalProperties": false,
+    "$defs": {
+        "Channel": {
```

Comment:
Is capitalizing defs the convention?

Reply:
No, I don't think so. I can change that if you like. 👍🏻

Reply:
If there's no benefit/convention then I have a weak preference for lowercase since my first impression is usually that the caps conveys some additional meaning.

Reply:
Done.